lib/msf/core/db_manager/service.rb

```diff
@@ -131,15 +140,31 @@ def report_service(opts)
   end
 
   # Returns a list of all services in the database
-  def services(wspace = workspace, only_up = false, proto = nil, addresses = nil, ports = nil, names = nil)
+  def services(opts)
+    opts.delete(:workspace) # Mdm::Service apparently doesn't have an upstream Mdm::Workspace association
+
+  ::ActiveRecord::Base.connection_pool.with_connection {
+    search_term = opts.delete(:search_term)
+    only_up = opts.delete(:only_up)
+    opts[:state] = [Msf::ServiceState::Open] if only_up
+    opts["hosts.address"] = opts.delete(:addresses)
+
+    opts.compact!
+    column_search_conditions = Msf::Util::DBManager.create_all_column_search_conditions(Mdm::Service, search_term)
```

Comment:
I overlooked this the other day, but if you look at the response to my comment below where you tested a modification using `arel_table`, you create an unnecessarily large query of empty regex comparisons when `search_term` isn't specified. Consider modifying this to something along the lines of:
```
if search_term && !search_term.empty?
  column_search_conditions = Msf::Util::DBManager.create_all_column_search_conditions(Mdm::Service, search_term)
  Mdm::Service.includes(:host).where(opts).where(column_search_conditions).order("hosts.address, port")
else
  Mdm::Service.includes(:host).where(opts).order("hosts.address, port")
end
```